public/app/features/dashboard/components/DashboardSettings/GeneralSettings.test.tsx

```diff
@@ -0,0 +1,119 @@
+import { DashboardModel } from '../../state';
+
+jest.mock('@grafana/runtime', () => ({
+  ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+  getBackendSrv: () => ({
+    search: jest.fn(() => [
+      { title: 'A', id: 'A' },
+      { title: 'B', id: 'B' },
+    ]),
+  }),
+}));
+
+jest.mock('app/core/services/context_srv', () => ({
+  contextSrv: {
+    user: { orgId: 1 },
+  },
+}));
+
+const setupTestContext = (options: Partial<Props>) => {
+  const defaults: Props = {
+    dashboard: ({
+      title: 'test dashboard title',
+      description: 'test dashboard description',
+      timepicker: {
+        refresh_intervals: ['5s', '10s', '30s', '1m', '5m', '15m', '30m', '1h', '2h', '1d', '2d'],
+        time_options: ['5m', '15m', '1h', '6h', '12h', '24h', '2d', '7d', '30d'],
+      },
+      meta: {
+        folderTitle: 'test',
+      },
+      timezone: 'utc',
+    } as unknown) as DashboardModel,
+    updateTimeZone: jest.fn(),
+  };
+
+  const props = { ...defaults, ...options };
+  const { rerender } = render(<GeneralSettings {...props} />);
+
+  return { rerender, props };
+};
+
+const clickSelectOption = async (selectElement: HTMLElement, optionText: string): Promise<void> => {
+  userEvent.click(byRole('textbox').get(selectElement));
+  userEvent.click(byText(optionText).get(selectElement));
+};
+
+describe('General Settings', () => {
+  describe('when component is mounted with timezone', () => {
+    it('should render correctly', () => {
+      setupTestContext({});
+      screen.getByDisplayValue('test dashboard title');
+      screen.getByDisplayValue('test dashboard description');
+      expect(screen.getByLabelText('Time zone picker select container')).toHaveTextContent(
+        'Coordinated Universal Time'
+      );
+    });
+  });
+  jest.mock('@grafana/runtime', () => ({
+    ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+    getBackendSrv: () => ({
+      search: jest.fn(() => [
+        { title: 'A', id: 'A' },
+        { title: 'B', id: 'B' },
+      ]),
+    }),
+  }));
+
+  jest.mock('app/core/services/context_srv', () => ({
+    contextSrv: {
+      user: { orgId: 1 },
+    },
+  }));
+  jest.mock('@grafana/runtime', () => ({
+    ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+    getBackendSrv: () => ({
+      search: jest.fn(() => [
+        { title: 'A', id: 'A' },
+        { title: 'B', id: 'B' },
+      ]),
+    }),
+  }));
+
+  jest.mock('app/core/services/context_srv', () => ({
+    contextSrv: {
+      user: { orgId: 1 },
+    },
+  }));
+  jest.mock('@grafana/runtime', () => ({
+    ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+    getBackendSrv: () => ({
+      search: jest.fn(() => [
+        { title: 'A', id: 'A' },
+        { title: 'B', id: 'B' },
+      ]),
+    }),
+  }));
+
+  jest.mock('app/core/services/context_srv', () => ({
+    contextSrv: {
+      user: { orgId: 1 },
+    },
+  }));
```

Comment:
These repeating mocks look weird?

Reply:
good catch! (_I just started using vim, this is what happened when one is learning the commands_ 😅 )

Reply:
![](https://media3.giphy.com/media/efm1NmEY9oRNY2EMp1/giphy-downsized-medium.gif)